packages/utils/src/internals/sitemap.ts

```diff
@@ -255,7 +259,7 @@ export async function* parseSitemap<T extends ParseSitemapOptions>(
                         },
                     );
 
-                    let error: Error | null = null;
+                    let error: { error: Error; type: 'fetch' | 'parser' } | null = null;
```

Comment:
what about a custom error class, with a `type` property on it? or this is just internal code?

Reply:
Yeah, this exception does not leave the stream.